clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -2938,36 +2914,17 @@ void CodeGenFunction::EmitRISCVMultiVersionResolver(
   bool HasDefault = false;
   unsigned DefaultIndex = 0;
 
-  SmallVector<CodeGenFunction::MultiVersionResolverOption, 10> CurrOptions(
-      Options);
-
-  llvm::stable_sort(
-      CurrOptions, [](const CodeGenFunction::MultiVersionResolverOption &LHS,
-                      const CodeGenFunction::MultiVersionResolverOption &RHS) {
-        return getPriorityFromAttrString(LHS.Conditions.Features[0]) >
-               getPriorityFromAttrString(RHS.Conditions.Features[0]);
-      });
-
   // Check the each candidate function.
-  for (unsigned Index = 0; Index < CurrOptions.size(); Index++) {
+  for (unsigned Index = 0; Index < Options.size(); Index++) {
 
-    if (CurrOptions[Index].Conditions.Features[0].starts_with("default")) {
+    if (Options[Index].Features.empty()) {
       HasDefault = true;
       DefaultIndex = Index;
       continue;
     }
 
     Builder.SetInsertPoint(CurBlock);
 
-    std::vector<std::string> TargetAttrFeats =
-        getContext()
-            .getTargetInfo()
-            .parseTargetAttr(CurrOptions[Index].Conditions.Features[0])
```

Comment:
The string wasn't parsed for the RISC-V target during the MultiVersionResolverOption, and there is a reason why the RISC-V target needs to postpone parseTargetAttr until emitting the resolver function. The 'priority=' attribute cannot be dropped before the MultiVersionResolverOption sorting. Before emitting the resolver function, it needs to normalize extension names to retrieve the mask bits for these extensions and remove unnecessary strings like 'arch=' and 'priority=<digits>'. 

The following statement
https://github.com/llvm/llvm-project/blob/c147f0cd439812114322e4edb72795d848af3d6d/clang/lib/CodeGen/CodeGenFunction.cpp#L2953
 seem to implement the similar feature like parseTargetAttr by dropping the "arch=" and "priority=". Why not directly invoke parseTargetAttr here instead of indirectly wrapping it as another hook?

> "arch=+zbb,+zba;priority=10" -> ["+zbb", "+zba"]

Reply:
I am using parseTargetAttr in latest revision, which had to be adapted for already delimited strings.